### PR TITLE
Include input stats for nodes derived from AbstractJoinNode

### PR DESCRIPTION
**Pull request by majetideepak**
Input stats are currently only printed for leaf nodes. This PR allows including input stats for Join nodes.
This PR also does a minor refactor of iterators of type `for (const auto& entry : stats.operatorStats)` to  `for (const auto& [opName, opStats] : stats.operatorStats)` inside `PlanNodeStats.cpp`

Output of printPlanWithStats before this change
```
HashJoin[INNER o_custkey=c_custkey]
Output: 244 rows (29626235 bytes), Cpu time: 809572544ns, Blocked wall time: 7670539000ns, Peak memory: 55574528 bytes
HashBuild: Output: 0 rows (0 bytes), Cpu time: 698417478ns, Blocked wall time: 1161888000ns, Peak memory: 54525952 bytes
HashProbe: Output: 244 rows (29626235 bytes), Cpu time: 111155066ns, Blocked wall time: 6508651000ns, Peak memory: 1048576 bytes
```
Output of printPlanWithStats after this change
```
HashJoin[INNER o_custkey=c_custkey]
Input: 1500624 rows (50725389 bytes), Raw Input: 0 rows (0 bytes), Output: 624 rows (76300877 bytes), Cpu time: 953904744ns, Blocked wall time: 11614288000ns, Peak memory: 38797312 bytes
HashBuild: Input: 1500000 rows (0 bytes), Raw Input: 0 rows (0 bytes), Output: 0 rows (0 bytes), Cpu time: 707924088ns, Blocked wall time: 891212000ns, Peak memory: 37748736 bytes
HashProbe: Input: 624 rows (50725389 bytes), Raw Input: 0 rows (0 bytes), Output: 624 rows (76300877 bytes), Cpu time: 245980656ns, Blocked wall time: 10723076000ns, Peak memory: 1048576 bytes
```

Note that `HashBuild` now is also easy to understand as we do not have to look for the corresponding operator feeding it.

### CODE REVIEW

**Comment by mbasmanova** [comment]
@majetideepak Deepak, it is not very clear to me what's the benefit of this change. Would you describe the problem you are solving? Perhaps, share the before and after outputs of the printPlanWithStats.

**Comment by majetideepak** [comment]
@mbasmanova I updated the description with the outputs of printPlanWithStats. Essentially, the problem this PR fixes is that the Input Stats are now also printed for Join nodes, which are helpful for understanding performance.

**Comment by mbasmanova** [comment]
@majetideepak Thank you for clarifying. I understand that you are saying that source node for HashBuild operator may be pretty far away from the HashBuild itself because there is the whole sub-tree for the HashProbe in between. Is that right? Showing the whole plan before and after would make that clearer.

"Raw Input" stats apply only to source / leaf nodes. It would be nice to avoid including these for non-source nodes.

It would be nice to generalize this to show input stats not only for Join operators but for all multi-operator plan nodes.

File: velox/exec/PlanNodeStats.cpp [hunk 1]
```diff
@@ -49,8 +49,8 @@ void PlanNodeStats::addTotals(const OperatorStats& stats) {
 
   peakMemoryBytes += stats.memoryStats.peakTotalMemoryReservation;
 
-  for (const auto& entry : stats.runtimeStats) {
-    customStats[entry.first].merge(entry.second);
+  for (const auto& [name, metric] : stats.runtimeStats) {
```
Comment (mbasmanova) [inline velox/exec/PlanNodeStats.cpp:52]: These refactors are nice, but they make it difficult to focus on the substantive changes. Any chance these can be moved into a separate PR?

**Comment by majetideepak** [comment]
@mbasmanova Your explanation is more apt. The Input to multi-node operators(like join) is somewhere in the graph. The idea here is to show all the input and output stats to make it easier to visualize.
It makes sense to generalize this to all multi-node operators. I will iterate on this PR. Thanks for the feedback.
